client/state/ui/editor/video-editor/reducer.js

```diff
@@ -0,0 +1,149 @@
+	VIDEO_EDITOR_VIDEO_HAS_LOADED,
+} from 'state/action-types';
+
+/**
+ * Tracks video loading state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action object
+ * @return {Object}        Updated state
+ */
+export const videoIsLoading = ( state = true, action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_VIDEO_HAS_LOADED:
+			return false;
+
+		case VIDEO_EDITOR_STATE_RESET:
+			return true;
+	}
+
+	return state;
+};
+
+/**
+ * Tracks poster updating state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action object
+ * @return {Object}        Updated state
+ */
+export const posterIsUpdating = ( state = false, action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_POSTER_UPDATE:
+			return true;
+
+		case VIDEO_EDITOR_POSTER_UPDATE_SUCCESS:
+		case VIDEO_EDITOR_POSTER_UPDATE_FAILURE:
+		case VIDEO_EDITOR_STATE_RESET:
+		case VIDEO_EDITOR_STATE_RESET_POSTER:
+			return false;
+	}
+
+	return state;
+};
+
+/**
+ * Tracks poster updated state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action object
+ * @return {Object}        Updated state
+ */
+export const posterIsUpdated = ( state = false, action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_POSTER_UPDATE_SUCCESS:
+		case VIDEO_EDITOR_POSTER_UPDATE_FAILURE:
+			return true;
+
+		case VIDEO_EDITOR_STATE_RESET:
+		case VIDEO_EDITOR_STATE_RESET_POSTER:
+			return false;
+	}
+
+	return state;
+};
+
+/**
+ * Tracks poster URL state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action object
+ * @return {Object}        Updated state
+ */
+export const poster = ( state = '', action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_POSTER_UPDATE_SUCCESS:
+			return action.poster;
+
+		case VIDEO_EDITOR_STATE_RESET:
+		case VIDEO_EDITOR_STATE_RESET_POSTER:
+			return '';
+	}
+
+	return state;
+};
+
+/**
+ * Tracks poster error state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action object
+ * @return {Object}        Updated state
+ */
+export const hasPosterUpdateError = ( state = false, action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_POSTER_UPDATE_FAILURE:
+			return true;
+
+		case VIDEO_EDITOR_POSTER_UPDATE:
+		case VIDEO_EDITOR_POSTER_UPDATE_SUCCESS:
+		case VIDEO_EDITOR_STATE_RESET:
+		case VIDEO_EDITOR_STATE_RESET_POSTER:
+			return false;
+	}
+
+	return state;
+};
+
+/**
+ * Tracks VideoPress script loading state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action object
+ * @return {Object}        Updated state
+ */
+export const hasScriptLoadError = ( state = false, action ) => {
+	switch ( action.type ) {
+		case VIDEO_EDITOR_SCRIPT_LOAD_ERROR:
+			return true;
+
+		case VIDEO_EDITOR_STATE_RESET:
+			return false;
+	}
+
+	return state;
+};
+
+export default combineReducers( {
+	hasPosterUpdateError,
+	hasScriptLoadError,
+	poster,
+	posterIsUpdating,
+	posterIsUpdated,
```

Comment:
@gwwar When the poster is being updated on the server, `posterIsUpdating` gets set and the UI will show a spinner.

I had to introduce a separate `posterIsUpdated` because I needed to know if the poster had been successfully updated, and if so, then close out the video editor and return to the previous screen - https://github.com/Automattic/wp-calypso/pull/11648/files#diff-35b626cb9b7af72412a7ab302d8fc3f6R67